[Layout foundations] Inline component prototype

aveline opened the pull request:
### WHY are these changes introduced?

Fixes https://github.com/Shopify/polaris/issues/6895

<!--
  Context about the problem that’s being addressed.
-->

### WHAT is this pull request doing?

Introduces the `Inline` component which can be used to manage spacing between inline elements

<img width="936" alt="Screen Shot 2022-08-25 at 2 45 31 PM" src="https://user-images.githubusercontent.com/3474483/186940873-9e24b30e-abd0-4288-8067-f1a7214d280b.png">

<details>
  <summary>Playground sample code</summary>

```
      <Text variant="bodySm" as="h3">
        default
      </Text>
      <Inline>
        <Text variant="heading4xl" as="h2">
          Inline
        </Text>
        <Badge>One</Badge>
        <Badge>Two</Badge>
        <Badge>Three</Badge>
        <Icon source={CapitalMajor} color="primary" />
      </Inline>
      <hr />
      <Text variant="bodySm" as="h3">
        alignY center
      </Text>
      <Inline alignY="center">
        <Text variant="heading4xl" as="h2">
          Inline
        </Text>
        <Badge>One</Badge>
        <Badge>Two</Badge>
        <Badge>Three</Badge>
        <Icon source={CapitalMajor} color="primary" />
      </Inline>
      <hr />
      <Text variant="bodySm" as="h3">
        alignY top
      </Text>
      <Inline alignY="top">
        <Text variant="heading4xl" as="h2">
          Inline
        </Text>
        <Badge>One</Badge>
        <Badge>Two</Badge>
        <Badge>Three</Badge>
        <Icon source={CapitalMajor} color="primary" />
      </Inline>
      <hr />
      <Text variant="bodySm" as="h3">
        alignY bottom
      </Text>
      <Inline alignY="bottom">
        <Text variant="heading4xl" as="h2">
          Inline
        </Text>
        <Badge>One</Badge>
        <Badge>Two</Badge>
        <Badge>Three</Badge>
        <Icon source={CapitalMajor} color="primary" />
      </Inline>
      <hr />
      <Text variant="bodySm" as="h3">
        alignY baseline
      </Text>
      <Inline alignY="baseline">
        <Text variant="heading4xl" as="h2">
          Inline
        </Text>
        <Badge>One</Badge>
        <Badge>Two</Badge>
        <Badge>Three</Badge>
        <Icon source={CapitalMajor} color="primary" />
      </Inline>
      <hr />
      <Text variant="bodySm" as="h3">
        align start
      </Text>
      <Inline align="start">
        <Text variant="heading4xl" as="h2">
          Inline
        </Text>
        <Badge>One</Badge>
        <Badge>Two</Badge>
        <Badge>Three</Badge>
        <Icon source={CapitalMajor} color="primary" />
      </Inline>
      <hr />
      <Text variant="bodySm" as="h3">
        align center
      </Text>
      <Inline align="center">
        <Text variant="heading4xl" as="h2">
          Inline
        </Text>
        <Badge>One</Badge>
        <Badge>Two</Badge>
        <Badge>Three</Badge>
        <Icon source={CapitalMajor} color="primary" />
      </Inline>
      <hr />
      <Text variant="bodySm" as="h3">
        align end
      </Text>
      <Inline align="end">
        <Text variant="heading4xl" as="h2">
          Inline
        </Text>
        <Badge>One</Badge>
        <Badge>Two</Badge>
        <Badge>Three</Badge>
        <Icon source={CapitalMajor} color="primary" />
      </Inline>
      <hr />
      <Text variant="bodySm" as="h3">
        align center alignY center
      </Text>
      <Inline align="center" alignY="center">
        <Text variant="heading4xl" as="h2">
          Inline
        </Text>
        <Badge>One</Badge>
        <Badge>Two</Badge>
        <Badge>Three</Badge>
        <Icon source={CapitalMajor} color="primary" />
      </Inline>
      <hr />
      <Text variant="bodySm" as="h3">
        spacing 0
      </Text>
      <Inline spacing="0">
        <Text variant="heading4xl" as="h2">
          Inline
        </Text>
        <Badge>One</Badge>
        <Badge>Two</Badge>
        <Badge>Three</Badge>
        <Icon source={CapitalMajor} color="primary" />
      </Inline>
      <hr />
      <Text variant="bodySm" as="h3">
        spacing 05
      </Text>
      <Inline spacing="05">
        <Text variant="heading4xl" as="h2">
          Inline
        </Text>
        <Badge>One</Badge>
        <Badge>Two</Badge>
        <Badge>Three</Badge>
        <Icon source={CapitalMajor} color="primary" />
      </Inline>
      <hr />
      <Text variant="bodySm" as="h3">
        spacing 4
      </Text>
      <Inline spacing="4">
        <Text variant="heading4xl" as="h2">
          Inline
        </Text>
        <Badge>One</Badge>
        <Badge>Two</Badge>
        <Badge>Three</Badge>
        <Icon source={CapitalMajor} color="primary" />
      </Inline>
      <hr />
      <Text variant="bodySm" as="h3">
        spacing 10
      </Text>
      <Inline spacing="10">
        <Text variant="heading4xl" as="h2">
          Inline
        </Text>
        <Badge>One</Badge>
        <Badge>Two</Badge>
        <Badge>Three</Badge>
        <Icon source={CapitalMajor} color="primary" />
      </Inline>
      <hr />
      <Text variant="bodySm" as="h3">
        spacing 32
      </Text>
      <Inline spacing="32">
        <Text variant="heading4xl" as="h2">
          Inline
        </Text>
        <Badge>One</Badge>
        <Badge>Two</Badge>
        <Badge>Three</Badge>
        <Icon source={CapitalMajor} color="primary" />
      </Inline>
```
  
</details>

aveline committed:
Add `Inline` component

aveline committed:
Add spacing values to `Inline` component

aveline committed:
Merge branch 'main' into layout-inline-component

aveline committed:
Fix typo

aveline committed:
Fix Inline stories

aveline committed:
Remove memoization

chazdean commented on polaris-react/src/components/Inline/Inline.scss:
```diff
@@ -0,0 +1,117 @@
+}
+
+.spacing1 {
+  --pc-stack-spacing: var(--p-space1);
```
You'll want to change this
```suggestion
  --pc-stack-spacing: var(--p-space-1);
```

laurkim committed:
[Layout foundations] Add alpha `Box` component (#7000)

Resolves #6892.

<!--
  Context about the problem that’s being addressed.
-->

Adds a new alpha `Box` component.
    <details>
      <summary>Using `Box` to recreate the `Card` component</summary>
      <img src="https://user-images.githubusercontent.com/26749317/187263883-0e8194f5-fbd8-4227-bdf5-66e868ffa573.png" alt="Using `Box` to recreate the `Card` component">
    </details>

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<!--
  Give as much information as needed to experiment with the component
  in the playground.
-->

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';

import {Page, Card, Text, Box} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
      <Box
        background="surface"
        borderRadius="radius-2"
        padding="5"
        shadow="card"
      >
        <Text as="h2" variant="headingMd">
          Online store dashboard
        </Text>
        <Text as="p" variant="bodyMd">
          This was made using the 📦 Box 📦 component.
        </Text>
      </Box>
      <div
        style={{
          borderTop: 'var(--p-border-divider)',
          marginTop: 'var(--p-space-8)',
          marginBottom: 'var(--p-space-8)',
        }}
      ></div>
      <Card title="Online store dashboard" sectioned>
        <Text as="p" variant="bodyMd">
          This was made using the 🃏 Card 🃏 component.
        </Text>
      </Card>
    </Page>
  );
}
```

</details>

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

Co-authored-by: Aaron Casanova <aaron.casanova@shopify.com>
Co-authored-by: Aveline Thelen <aveline.thelen@shopify.com>
Co-authored-by: Alex Page <alex.page@shopify.com>
Co-authored-by: Kyle Durand <kyledurand@users.noreply.github.com>

laurkim committed:
[Layout foundations] Add alpha `Box` component (#7000)

Resolves #6892.

<!--
  Context about the problem that’s being addressed.
-->

Adds a new alpha `Box` component.
    <details>
      <summary>Using `Box` to recreate the `Card` component</summary>
      <img src="https://user-images.githubusercontent.com/26749317/187263883-0e8194f5-fbd8-4227-bdf5-66e868ffa573.png" alt="Using `Box` to recreate the `Card` component">
    </details>

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<!--
  Give as much information as needed to experiment with the component
  in the playground.
-->

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';

import {Page, Card, Text, Box} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
      <Box
        background="surface"
        borderRadius="radius-2"
        padding="5"
        shadow="card"
      >
        <Text as="h2" variant="headingMd">
          Online store dashboard
        </Text>
        <Text as="p" variant="bodyMd">
          This was made using the 📦 Box 📦 component.
        </Text>
      </Box>
      <div
        style={{
          borderTop: 'var(--p-border-divider)',
          marginTop: 'var(--p-space-8)',
          marginBottom: 'var(--p-space-8)',
        }}
      ></div>
      <Card title="Online store dashboard" sectioned>
        <Text as="p" variant="bodyMd">
          This was made using the 🃏 Card 🃏 component.
        </Text>
      </Card>
    </Page>
  );
}
```

</details>

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

Co-authored-by: Aaron Casanova <aaron.casanova@shopify.com>
Co-authored-by: Aveline Thelen <aveline.thelen@shopify.com>
Co-authored-by: Alex Page <alex.page@shopify.com>
Co-authored-by: Kyle Durand <kyledurand@users.noreply.github.com>

laurkim committed:
[Layout foundations] Add alpha `Box` component (#7000)

Resolves #6892.

<!--
  Context about the problem that’s being addressed.
-->

Adds a new alpha `Box` component.
    <details>
      <summary>Using `Box` to recreate the `Card` component</summary>
      <img src="https://user-images.githubusercontent.com/26749317/187263883-0e8194f5-fbd8-4227-bdf5-66e868ffa573.png" alt="Using `Box` to recreate the `Card` component">
    </details>

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<!--
  Give as much information as needed to experiment with the component
  in the playground.
-->

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';

import {Page, Card, Text, Box} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
      <Box
        background="surface"
        borderRadius="radius-2"
        padding="5"
        shadow="card"
      >
        <Text as="h2" variant="headingMd">
          Online store dashboard
        </Text>
        <Text as="p" variant="bodyMd">
          This was made using the 📦 Box 📦 component.
        </Text>
      </Box>
      <div
        style={{
          borderTop: 'var(--p-border-divider)',
          marginTop: 'var(--p-space-8)',
          marginBottom: 'var(--p-space-8)',
        }}
      ></div>
      <Card title="Online store dashboard" sectioned>
        <Text as="p" variant="bodyMd">
          This was made using the 🃏 Card 🃏 component.
        </Text>
      </Card>
    </Page>
  );
}
```

</details>

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

Co-authored-by: Aaron Casanova <aaron.casanova@shopify.com>
Co-authored-by: Aveline Thelen <aveline.thelen@shopify.com>
Co-authored-by: Alex Page <alex.page@shopify.com>
Co-authored-by: Kyle Durand <kyledurand@users.noreply.github.com>

laurkim committed:
[Layout foundations] Refactor `Box` without polymorphic dep (#7062)

Co-authored-by: Kyle Durand <kyle.durand@shopify.com>

laurkim committed:
[Layout foundations] Refactor `Box` without polymorphic dep (#7062)

Co-authored-by: Kyle Durand <kyle.durand@shopify.com>

laurkim committed:
[Layout foundations] Refactor `Box` without polymorphic dep (#7062)

Co-authored-by: Kyle Durand <kyle.durand@shopify.com>

kyledurand committed:
[Layout foundations] Add `Columns` component (#7057)

### WHY are these changes introduced?

Fixes https://github.com/Shopify/polaris/issues/6897

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';

import {Page, Columns, Stack} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      <Stack vertical>
        <h2>Equal columns example</h2>
        <Columns
          columns={{xs: 2, sm: '1fr 1.5fr', md: 4, lg: 6}}
          gap={{xs: '1', lg: '4'}}
        >
          <div style={{background: 'aquamarine'}}>one</div>
          <div style={{background: 'aquamarine'}}>two</div>
          <div style={{background: 'aquamarine'}}>three</div>
          <div style={{background: 'aquamarine'}}>four</div>
          <div style={{background: 'aquamarine'}}>five</div>
          <div style={{background: 'aquamarine'}}>six</div>
        </Columns>

        <h2>Non equal columns example</h2>
        <Columns
          columns={{
            xs: '1.5fr 0.5fr',
            sm: '2fr 1fr',
            md: '1fr 3fr auto 1fr',
            lg: '1fr 4fr auto 2fr 3fr auto',
          }}
          gap={{xs: '4'}}
        >
          <div style={{background: 'aquamarine'}}>one</div>
          <div style={{background: 'aquamarine'}}>two</div>
          <div style={{background: 'aquamarine'}}>three</div>
          <div style={{background: 'aquamarine'}}>four</div>
          <div style={{background: 'aquamarine'}}>five</div>
          <div style={{background: 'aquamarine'}}>six</div>
        </Columns>
      </Stack>
    </Page>
  );
}

```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

Co-authored-by: Lo Kim <lo.kim@shopify.com>

kyledurand committed:
[Layout foundations] Add `Columns` component (#7057)

### WHY are these changes introduced?

Fixes https://github.com/Shopify/polaris/issues/6897

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';

import {Page, Columns, Stack} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      <Stack vertical>
        <h2>Equal columns example</h2>
        <Columns
          columns={{xs: 2, sm: '1fr 1.5fr', md: 4, lg: 6}}
          gap={{xs: '1', lg: '4'}}
        >
          <div style={{background: 'aquamarine'}}>one</div>
          <div style={{background: 'aquamarine'}}>two</div>
          <div style={{background: 'aquamarine'}}>three</div>
          <div style={{background: 'aquamarine'}}>four</div>
          <div style={{background: 'aquamarine'}}>five</div>
          <div style={{background: 'aquamarine'}}>six</div>
        </Columns>

        <h2>Non equal columns example</h2>
        <Columns
          columns={{
            xs: '1.5fr 0.5fr',
            sm: '2fr 1fr',
            md: '1fr 3fr auto 1fr',
            lg: '1fr 4fr auto 2fr 3fr auto',
          }}
          gap={{xs: '4'}}
        >
          <div style={{background: 'aquamarine'}}>one</div>
          <div style={{background: 'aquamarine'}}>two</div>
          <div style={{background: 'aquamarine'}}>three</div>
          <div style={{background: 'aquamarine'}}>four</div>
          <div style={{background: 'aquamarine'}}>five</div>
          <div style={{background: 'aquamarine'}}>six</div>
        </Columns>
      </Stack>
    </Page>
  );
}

```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

Co-authored-by: Lo Kim <lo.kim@shopify.com>

kyledurand committed:
[Layout foundations] Add `Columns` component (#7057)

### WHY are these changes introduced?

Fixes https://github.com/Shopify/polaris/issues/6897

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';

import {Page, Columns, Stack} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      <Stack vertical>
        <h2>Equal columns example</h2>
        <Columns
          columns={{xs: 2, sm: '1fr 1.5fr', md: 4, lg: 6}}
          gap={{xs: '1', lg: '4'}}
        >
          <div style={{background: 'aquamarine'}}>one</div>
          <div style={{background: 'aquamarine'}}>two</div>
          <div style={{background: 'aquamarine'}}>three</div>
          <div style={{background: 'aquamarine'}}>four</div>
          <div style={{background: 'aquamarine'}}>five</div>
          <div style={{background: 'aquamarine'}}>six</div>
        </Columns>

        <h2>Non equal columns example</h2>
        <Columns
          columns={{
            xs: '1.5fr 0.5fr',
            sm: '2fr 1fr',
            md: '1fr 3fr auto 1fr',
            lg: '1fr 4fr auto 2fr 3fr auto',
          }}
          gap={{xs: '4'}}
        >
          <div style={{background: 'aquamarine'}}>one</div>
          <div style={{background: 'aquamarine'}}>two</div>
          <div style={{background: 'aquamarine'}}>three</div>
          <div style={{background: 'aquamarine'}}>four</div>
          <div style={{background: 'aquamarine'}}>five</div>
          <div style={{background: 'aquamarine'}}>six</div>
        </Columns>
      </Stack>
    </Page>
  );
}

```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

Co-authored-by: Lo Kim <lo.kim@shopify.com>

chazdean committed:
[Layout foundations] Stack component prototype (#7036)

<!--
  ☝️How to write a good PR title:
  - Prefix it with [ComponentName] (if applicable), for example: [Button]
  - Start with a verb, for example: Add, Delete, Improve, Fix…
  - Give as much context as necessary and as little as possible
  - Prefix it with [WIP] while it’s a work in progress
-->

### WHY are these changes introduced?

Fixes #6894  <!-- link to issue if one exists -->

<!--
  Context about the problem that’s being addressed.
-->

### WHAT is this pull request doing?

Adds a new alpha `Stack` component

 <details>
      <summary>Alignment</summary>
     <img width="977" alt="Screen Shot 2022-08-30 at 11 00 25 AM" src="https://user-images.githubusercontent.com/59836805/187471756-15963459-54fd-403d-93e2-9e7f0932865c.png">
    </details>

<details>
      <summary>Spacing</summary>
     <img width="978" alt="Screen Shot 2022-08-30 at 11 01 34 AM" src="https://user-images.githubusercontent.com/59836805/187471804-29d50f45-f00b-4007-abc9-575555fe37c5.png">
</details>

<!--
  Summary of the changes committed.

  Before / after screenshots are appreciated for UI changes. Make sure to include alt text that describes the screenshot.

  If you include an animated gif showing your change, wrapping it in a details tag is recommended. Gifs usually autoplay, which can cause accessibility issues for people reviewing your PR:

    <details>
      <summary>Summary of your gif(s)</summary>
      <img src="..." alt="Description of what the gif shows">
    </details>
-->

<!-- ℹ️ Delete the following for small / trivial changes -->

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<!--
  Give as much information as needed to experiment with the component
  in the playground.
-->

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';

import {Page, AlphaStack, Badge, Text} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      <AlphaStack>
        <Badge status="attention">start</Badge>
        <Badge>Processing</Badge>
        <Badge>Fulfilled</Badge>
      </AlphaStack>
      <hr />
      <AlphaStack align="center">
        <Badge status="attention">center</Badge>
        <Badge>Processing</Badge>
        <Badge>Fulfilled</Badge>
      </AlphaStack>
      <hr />
      <AlphaStack align="end">
        <Badge status="attention">end</Badge>
        <Badge>Processing</Badge>
        <Badge>Fulfilled</Badge>
      </AlphaStack>
      <hr />
      <AlphaStack spacing="0">
        <Badge status="attention">spacing 0</Badge>
        <Badge>Processing</Badge>
        <Badge>Fulfilled</Badge>
      </AlphaStack>
      <hr />
      <AlphaStack spacing="4">
        <Badge status="attention">spacing 4</Badge>
        <Badge>Processing</Badge>
        <Badge>Fulfilled</Badge>
      </AlphaStack>
      <hr />
      <AlphaStack spacing="10">
        <Badge status="attention">spacing 10</Badge>
        <Badge>Processing</Badge>
        <Badge>Fulfilled</Badge>
      </AlphaStack>
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

Co-authored-by: Lo Kim <lo.kim@shopify.com>

chazdean committed:
[Layout foundations] Stack component prototype (#7036)

<!--
  ☝️How to write a good PR title:
  - Prefix it with [ComponentName] (if applicable), for example: [Button]
  - Start with a verb, for example: Add, Delete, Improve, Fix…
  - Give as much context as necessary and as little as possible
  - Prefix it with [WIP] while it’s a work in progress
-->

### WHY are these changes introduced?

Fixes #6894  <!-- link to issue if one exists -->

<!--
  Context about the problem that’s being addressed.
-->

### WHAT is this pull request doing?

Adds a new alpha `Stack` component

 <details>
      <summary>Alignment</summary>
     <img width="977" alt="Screen Shot 2022-08-30 at 11 00 25 AM" src="https://user-images.githubusercontent.com/59836805/187471756-15963459-54fd-403d-93e2-9e7f0932865c.png">
    </details>

<details>
      <summary>Spacing</summary>
     <img width="978" alt="Screen Shot 2022-08-30 at 11 01 34 AM" src="https://user-images.githubusercontent.com/59836805/187471804-29d50f45-f00b-4007-abc9-575555fe37c5.png">
</details>

<!--
  Summary of the changes committed.

  Before / after screenshots are appreciated for UI changes. Make sure to include alt text that describes the screenshot.

  If you include an animated gif showing your change, wrapping it in a details tag is recommended. Gifs usually autoplay, which can cause accessibility issues for people reviewing your PR:

    <details>
      <summary>Summary of your gif(s)</summary>
      <img src="..." alt="Description of what the gif shows">
    </details>
-->

<!-- ℹ️ Delete the following for small / trivial changes -->

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<!--
  Give as much information as needed to experiment with the component
  in the playground.
-->

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';

import {Page, AlphaStack, Badge, Text} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      <AlphaStack>
        <Badge status="attention">start</Badge>
        <Badge>Processing</Badge>
        <Badge>Fulfilled</Badge>
      </AlphaStack>
      <hr />
      <AlphaStack align="center">
        <Badge status="attention">center</Badge>
        <Badge>Processing</Badge>
        <Badge>Fulfilled</Badge>
      </AlphaStack>
      <hr />
      <AlphaStack align="end">
        <Badge status="attention">end</Badge>
        <Badge>Processing</Badge>
        <Badge>Fulfilled</Badge>
      </AlphaStack>
      <hr />
      <AlphaStack spacing="0">
        <Badge status="attention">spacing 0</Badge>
        <Badge>Processing</Badge>
        <Badge>Fulfilled</Badge>
      </AlphaStack>
      <hr />
      <AlphaStack spacing="4">
        <Badge status="attention">spacing 4</Badge>
        <Badge>Processing</Badge>
        <Badge>Fulfilled</Badge>
      </AlphaStack>
      <hr />
      <AlphaStack spacing="10">
        <Badge status="attention">spacing 10</Badge>
        <Badge>Processing</Badge>
        <Badge>Fulfilled</Badge>
      </AlphaStack>
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

Co-authored-by: Lo Kim <lo.kim@shopify.com>

chazdean committed:
[Layout foundations] Stack component prototype (#7036)

<!--
  ☝️How to write a good PR title:
  - Prefix it with [ComponentName] (if applicable), for example: [Button]
  - Start with a verb, for example: Add, Delete, Improve, Fix…
  - Give as much context as necessary and as little as possible
  - Prefix it with [WIP] while it’s a work in progress
-->

### WHY are these changes introduced?

Fixes #6894  <!-- link to issue if one exists -->

<!--
  Context about the problem that’s being addressed.
-->

### WHAT is this pull request doing?

Adds a new alpha `Stack` component

 <details>
      <summary>Alignment</summary>
     <img width="977" alt="Screen Shot 2022-08-30 at 11 00 25 AM" src="https://user-images.githubusercontent.com/59836805/187471756-15963459-54fd-403d-93e2-9e7f0932865c.png">
    </details>

<details>
      <summary>Spacing</summary>
     <img width="978" alt="Screen Shot 2022-08-30 at 11 01 34 AM" src="https://user-images.githubusercontent.com/59836805/187471804-29d50f45-f00b-4007-abc9-575555fe37c5.png">
</details>

<!--
  Summary of the changes committed.

  Before / after screenshots are appreciated for UI changes. Make sure to include alt text that describes the screenshot.

  If you include an animated gif showing your change, wrapping it in a details tag is recommended. Gifs usually autoplay, which can cause accessibility issues for people reviewing your PR:

    <details>
      <summary>Summary of your gif(s)</summary>
      <img src="..." alt="Description of what the gif shows">
    </details>
-->

<!-- ℹ️ Delete the following for small / trivial changes -->

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<!--
  Give as much information as needed to experiment with the component
  in the playground.
-->

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';

import {Page, AlphaStack, Badge, Text} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      <AlphaStack>
        <Badge status="attention">start</Badge>
        <Badge>Processing</Badge>
        <Badge>Fulfilled</Badge>
      </AlphaStack>
      <hr />
      <AlphaStack align="center">
        <Badge status="attention">center</Badge>
        <Badge>Processing</Badge>
        <Badge>Fulfilled</Badge>
      </AlphaStack>
      <hr />
      <AlphaStack align="end">
        <Badge status="attention">end</Badge>
        <Badge>Processing</Badge>
        <Badge>Fulfilled</Badge>
      </AlphaStack>
      <hr />
      <AlphaStack spacing="0">
        <Badge status="attention">spacing 0</Badge>
        <Badge>Processing</Badge>
        <Badge>Fulfilled</Badge>
      </AlphaStack>
      <hr />
      <AlphaStack spacing="4">
        <Badge status="attention">spacing 4</Badge>
        <Badge>Processing</Badge>
        <Badge>Fulfilled</Badge>
      </AlphaStack>
      <hr />
      <AlphaStack spacing="10">
        <Badge status="attention">spacing 10</Badge>
        <Badge>Processing</Badge>
        <Badge>Fulfilled</Badge>
      </AlphaStack>
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md) changes in the style guide

Co-authored-by: Lo Kim <lo.kim@shopify.com>

laurkim committed:
[Layout foundations] Add tests for `Box` (#7094)

laurkim committed:
[Layout foundations] Add tests for `Box` (#7094)

laurkim committed:
[Layout foundations] Add tests for `Box` (#7094)

laurkim committed:
[Layout foundations] Remove `display` from root `Box` styling

laurkim committed:
[Layout foundations] Remove `display` from root `Box` styling

laurkim committed:
[Layout foundations] Remove `display` from root `Box` styling

laurkim committed:
Fix `polaris-icons` package versions

laurkim committed:
Fix `polaris-icons` package versions

laurkim committed:
Fix `polaris-icons` package versions

chazdean committed:
[Layout foundations] Add tests for Stack (#7106)

<!--
  ☝️How to write a good PR title:
- Prefix it with [ComponentName] (if applicable), for example: [Button]
  - Start with a verb, for example: Add, Delete, Improve, Fix…
  - Give as much context as necessary and as little as possible
  - Prefix it with [WIP] while it’s a work in progress
-->

### WHY are these changes introduced?

Resolves #7093 <!-- link to issue if one exists -->

<!--
  Context about the problem that’s being addressed.
-->

### WHAT is this pull request doing?

Adds test for the `AlphaStack` component

<!--
  Summary of the changes committed.

Before / after screenshots are appreciated for UI changes. Make sure to
include alt text that describes the screenshot.

If you include an animated gif showing your change, wrapping it in a
details tag is recommended. Gifs usually autoplay, which can cause
accessibility issues for people reviewing your PR:

    <details>
      <summary>Summary of your gif(s)</summary>
      <img src="..." alt="Description of what the gif shows">
    </details>
-->

<!-- ℹ️ Delete the following for small / trivial changes -->

### How to 🎩

🖥 [Local development
instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting
guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog
guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<!--
  Give as much information as needed to experiment with the component
  in the playground.
-->

<details>
<summary>Copy-paste this code in
<code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on
[mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple
browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for
[accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted
documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md)
changes in the style guide

chazdean committed:
[Layout foundations] Add tests for Stack (#7106)

<!--
  ☝️How to write a good PR title:
- Prefix it with [ComponentName] (if applicable), for example: [Button]
  - Start with a verb, for example: Add, Delete, Improve, Fix…
  - Give as much context as necessary and as little as possible
  - Prefix it with [WIP] while it’s a work in progress
-->

### WHY are these changes introduced?

Resolves #7093 <!-- link to issue if one exists -->

<!--
  Context about the problem that’s being addressed.
-->

### WHAT is this pull request doing?

Adds test for the `AlphaStack` component

<!--
  Summary of the changes committed.

Before / after screenshots are appreciated for UI changes. Make sure to
include alt text that describes the screenshot.

If you include an animated gif showing your change, wrapping it in a
details tag is recommended. Gifs usually autoplay, which can cause
accessibility issues for people reviewing your PR:

    <details>
      <summary>Summary of your gif(s)</summary>
      <img src="..." alt="Description of what the gif shows">
    </details>
-->

<!-- ℹ️ Delete the following for small / trivial changes -->

### How to 🎩

🖥 [Local development
instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting
guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog
guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<!--
  Give as much information as needed to experiment with the component
  in the playground.
-->

<details>
<summary>Copy-paste this code in
<code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on
[mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple
browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for
[accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted
documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md)
changes in the style guide

chazdean committed:
[Layout foundations] Add tests for Stack (#7106)

<!--
  ☝️How to write a good PR title:
- Prefix it with [ComponentName] (if applicable), for example: [Button]
  - Start with a verb, for example: Add, Delete, Improve, Fix…
  - Give as much context as necessary and as little as possible
  - Prefix it with [WIP] while it’s a work in progress
-->

### WHY are these changes introduced?

Resolves #7093 <!-- link to issue if one exists -->

<!--
  Context about the problem that’s being addressed.
-->

### WHAT is this pull request doing?

Adds test for the `AlphaStack` component

<!--
  Summary of the changes committed.

Before / after screenshots are appreciated for UI changes. Make sure to
include alt text that describes the screenshot.

If you include an animated gif showing your change, wrapping it in a
details tag is recommended. Gifs usually autoplay, which can cause
accessibility issues for people reviewing your PR:

    <details>
      <summary>Summary of your gif(s)</summary>
      <img src="..." alt="Description of what the gif shows">
    </details>
-->

<!-- ℹ️ Delete the following for small / trivial changes -->

### How to 🎩

🖥 [Local development
instructions](https://github.com/Shopify/polaris/blob/main/README.md#local-development)
🗒 [General tophatting
guidelines](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md)
📄 [Changelog
guidelines](https://github.com/Shopify/polaris/blob/main/.github/CONTRIBUTING.md#changelog)

<!--
  Give as much information as needed to experiment with the component
  in the playground.
-->

<details>
<summary>Copy-paste this code in
<code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on
[mobile](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple
browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers)
- [ ] Tested for
[accessibility](https://github.com/Shopify/polaris/blob/main/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted
documentation](https://github.com/Shopify/polaris/blob/main/documentation/Tophatting%20documentation.md)
changes in the style guide

aveline committed:
Update polaris-react/src/components/Inline/Inline.scss

Co-authored-by: Chaz Dean <59836805+chazdean@users.noreply.github.com>

aveline committed:
Merge branch 'layout-foundations-prototype' into layout-inline-component

aveline committed:
Refactor and remove child Item component

aveline committed:
Update Inline.test.tsx

aveline committed:
Merge branch 'layout-foundations-prototype' into layout-inline-component

aveline committed:
Merge branch 'layout-foundations-prototype' into layout-inline-component

aveline committed:
Create fresh-dingos-press.md

aveline committed:
Bump size limit (#7158)

### WHY are these changes introduced?
Bumps size-limit for `polaris-react-cjs` from `208kb` to `209kb` to
resolve failing CI checks.

aveline committed:
Merge branch 'layout-foundations-prototype' into layout-inline-component

kyledurand commented on polaris-react/src/components/Inline/Inline.tsx:
```diff
@@ -0,0 +1,56 @@
+    spacing && styles[variationName('spacing', spacing)],
+    align && styles[variationName('align', align)],
+    alignY && styles[variationName('alignY', alignY)],
+    wrap === false && styles.noWrap,
```
I think most of the css here can be replaced by setting the custom property to the wrapper i.e. 

`return <div className={styles.Inline} style={style}>{itemMarkup}</div>;`

where style would be similar to https://github.com/Shopify/polaris/blob/layout-foundations-prototype/polaris-react/src/components/Columns/Columns.tsx#L27-L38

Back when we had atoms it was fine but now we risk blowing up our bundle size with one off classes

aveline replied:
Yeah I think it would be great to explore some kind of shared utility or atomic classes for this. I would almost rather leave this in as a reminder to do so. Setting the custom property inline seems less legible to me.

kyledurand replied:
I'm curious how it's less legible. I've always found things like our classname variations hard to understand what's happening: `spacing && styles[variationName('spacing', spacing)]`

I've just done a [quick refactor](https://github.com/Shopify/polaris/pull/7174/files#diff-fa99a8ea2c4165b404905aedebdd75053975846cf7860bf30fda7a2150b7fd70R43-R48), it's very messy 😪 but I think it's nice and readable and brings the css down to ~7 lines.

Also GH copilot almost automatically brought the playground over into stories 🤯. All I did was keep pressing enter

kyledurand replied:
In terms of getting into atomic classes. I'm very hesitant to do so, at least right now. We'd have to generate them from tokens and consume them in Polaris react. This means recreating a subset of vanilla extract functionality which is probably not easy. We'd also need some integration tests between the two  packages. 

What we're left with there is still more css and increased bundle, and documenting how to use these utility classes.

Something like `import spacingStyles from '@shopify/polaris-tokens'` 

But we can't directly match our spacing system to css because numbers so we'd have to prefix them, I think.

`const className = spacingStyles.space05`

I'm rambling but in my mind it seems like more hassle than it's worth than to use tokens directly

aveline replied:
Thanks for the refactor, I see the value in reducing the css file size. In terms of legibility, I meant adding all that inline in the dom. It isn't too bad with only a few custom properties, but using this pattern where there could be several more, it will start to look messy. Maybe it's just a personal preference. Regardless, I see your point that this is a larger issue that we shouldn't be trying to solve in the layout work. So I'm on board with going with this pattern for now.

laurkim commented on .changeset/fresh-dingos-press.md:
```diff
@@ -0,0 +1,5 @@
+'@shopify/polaris': minor
+---
+
+Add alpha `Inline` component
```
```suggestion
Added alpha `Inline` component
```

aveline committed:
Update .changeset/fresh-dingos-press.md

Co-authored-by: Lo Kim <lo.kim@shopify.com>

kyledurand commented on polaris-react/src/components/Inline/Inline.scss:
```diff
@@ -0,0 +1,113 @@
+.Item {
+  flex: 0 0 auto;
+  min-width: 0;
+  max-width: 100%;
```
I'm not sure we need the min-width / max-width flex box fixes these days. Can we omit them until we notice weirdness? Also `flex: 0 0 auto` feels like it might be a bit too prescriptive but I'm open to it if you feel it's better DX

aveline replied:
Good call on all accounts. I think the `min-width` fix doesn't come into play unless there's some nested flex going on, so might not be necessary for this component. I'm good to remove unless we notice issues.

kyledurand reviewed (commented):
This is looking good Aveline! My main concern is getting the total lines of css down as much as possible but overall I think this is looking great

laurkim reviewed (approved):
LGTM! 🚀 💯 

My only suggestion is to reduce the class names in the `Inline.scss` file by using `style={style}` as @kyledurand suggested 👍

chazdean reviewed (approved):
🚀!

aveline committed:
Merge branch 'layout-foundations-prototype' into layout-inline-component

aveline committed:
Remove unnecessary css

kyledurand commented on polaris-react/src/components/Inline/Inline.scss:
```diff
@@ -0,0 +1,107 @@
+.Inline {
+  --pc-stack-spacing: var(--p-space-4);
```
```suggestion
  --pc-inline-spacing: var(--p-space-4);
```

kyledurand commented on polaris-react/src/components/Inline/Inline.scss:
```diff
@@ -0,0 +1,107 @@
+  display: flex;
+  flex-wrap: wrap;
+  align-items: stretch;
+  gap: var(--pc-stack-spacing);
```
```suggestion
  gap: var(--pc-inline-spacing);
```

kyledurand reviewed (approved):
Feel free to ship as is and iterate as you see fit 👍

kyledurand committed:
Refactor (#7174)

Co-authored-by: aveline <aveline@users.noreply.github.com>
